src/lib/components/language/Language.svelte

```diff
@@ -1,27 +1,42 @@
 <script>
     import { isLanguageMenuVisible } from './languageStore';
     import languageIcon from '../../../assets/icons/language.svg';
+    import ClickOutside from '../customEvent/ClickOutside.svelte';
+    import { hideMenu } from './languageHelper';
+    import LanguageInputs from './LanguageInputs.svelte';
     import { language } from '../../../store/global';
-    import { showMenu, hideMenu } from './languageHelper';
+
+    function toggleMenuVisibility() {
+        $isLanguageMenuVisible = !$isLanguageMenuVisible;
+    }
 </script>
 
 <div class="simple-signer language-container">
-    <div
-        tabindex="-1"
-        on:focus={showMenu}
-        on:blur={hideMenu}
-        class="simple-signer language-container-icon  {$isLanguageMenuVisible ? 'active' : ''}"
-    >
-        <img class="simple-signer language-icon" src={languageIcon} alt="*" />
-
-        <div class="simple-signer language-selector-container {$isLanguageMenuVisible ? '' : 'hidden'}">
-            <button class="language-active">{$language.ENGLISH}</button>
-            <button class="default">{$language.SPANISH}</button>
+    <ClickOutside on:clickoutside={hideMenu}>
+        <div class="simple-signer language-container-icon  {$isLanguageMenuVisible ? 'active' : ''}">
+            <button on:click={toggleMenuVisibility} class="simple-signer invisible-button">
+                <img class="simple-signer language-icon" src={languageIcon} alt="*" />
+            </button>
+            <div class="simple-signer language-selector-container {$isLanguageMenuVisible ? '' : 'hidden'}">
+                <LanguageInputs
+                    languageInputsProps={[
+                        { iso: 'es', text: $language.SPANISH },
+                        { iso: 'en', text: $language.ENGLISH },
```

Comment:
`'es'` and `'en'` should probably go in the $language structure, perhaps SPANISH_ISO and ENGLISH_ISO?